homeassistant/components/bluetooth/models.py

```diff
@@ -70,6 +70,20 @@ class BaseHaScanner:
     def discovered_devices(self) -> list[BLEDevice]:
         """Return a list of discovered devices."""
 
+    async def async_diagnostics(self) -> dict[str, Any]:
```

Comment:
Why is this a coroutine function? Will some scanners require awaiting here in the future?

Reply:
Yes. Once we have active remotes, I expect we will need to await

Reply:
Ok.